### ISSUE TITLE
Include other useful transforms from Vega

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#3264]
## Definitely useful

- [ ] fold
- [ ] sample
- [ ] sequence
- [ ] stack

## Probably useful? 

- [ ] project

**Comment by domoritz** [comment]
Sequence would be amazing for creating examples of mathematical functions. However, we have to decide whether it should go in data or transforms.

**Comment by kanitw** [comment]
To me, sequence feels more natural as a special type of data source.

**Comment by domoritz** [comment]
Yes, I think so too. It's just that Vega has it as a transform and I wanted to point out that it's not the most natural place and we can change that in Vega-Lite. Yet another reason to not just support all Vega transforms.

**Comment by domoritz** [comment]
Sequence is covered in https://github.com/vega/vega-lite/issues/3864

**Comment by domoritz** [comment]
Graticule is in https://github.com/vega/vega-lite/issues/3337

**Comment by domoritz** [comment]
I don't think we need pivot in Vega-Lite, do we?

**Comment by kanitw** [comment]
It's actually very useful.   For example, with `pivot`, we can create ["Multi Series Line Chart with Tooltip"](https://vega.github.io/vega-lite/examples/interactive_multi_line_tooltip.html) with the stocks  data (as the stocks data has a  long format, but we  need a pivoted format for "Multi Series Line Chart with Tooltip".

**Comment by pjsier** [comment]
@kanitw I'm interested in submitting a PR to add `pivot`, but from a quick look at the code I'm not sure how to implement `producedFields` since they're generated from the data values. Should it just return an empty set, or would that cause issues?

**Comment by domoritz** [comment]
@pjsier Thanks for looking into this. Figuring out the correct `producedFields` is super tricky here. I think we can only return `null` or `{}` in this case but we need to update the checks that use this field so that when we check for intersections and we have a `pivot` transform, we need to always say that there is an intersection. This will make it difficult to optimize the dataflow when there is a pivot transform.

**Comment by pjsier** [comment]
@domoritz sounds good, thanks! I’ll try to put together a PR with that setup

**Comment by kanitw** [comment]
"null" or "{}" would seems like "nothing".   I wonder if we should use a specific string literal "unknown" to make it explicit that we don't know at compile time.
